pkg/datarepair/repairer/repairer.go

```diff
@@ -135,3 +161,32 @@ func (service *Service) process(ctx context.Context) error {
 		})
 	}
 }
+
+// irrepairProcess picks items from irrepairabledb and spawns a repair worker
+func (service *Service) irrepairProcess(ctx context.Context) error {
+	limit := 1
+	var offset int64
+	for {
+		seg, err := service.irrdb.GetLimited(ctx, limit, offset)
+		if err != nil {
+			return err
+		}
+
+		// irrepairabledb empty
+		if len(seg) == 0 {
+			return nil
+		}
+
+		service.Limiter.Go(ctx, func() {
+			err := service.repairer.Repair(ctx, string(seg[0].GetPath()))
```

Comment:
It seems like we should be calling checker in general instead of repairer.

Reply:
I am bit unclear. This is my thought process. Using the analogy of classic example, producer - consumer. 

Checker is a producer. Checker's task is to iterate thru pointerDB and distribute the pieces to two other DBs namely RepairDB and IrreparableDB. There are two processes that run parallelly to consume the entires entered by the checker into these two DBs. 

The Repair process task is to repairs the entries entered that are in repairDB. Handle these scenarios:
1. make an assessment at the time of actual repair to see what piece number are actually lost. 
    1.1. At the time of entry and at the time of actual repair, the injured segments number COULD be different. So repair the actual segments that needed to be repaired and not that were entered at some point in past time ( the assumption I made is there is a constant churn rate of nodes)
   1.2 At the time of repair, a repair segment could become irreparable or doesn't need repair.
   
The Irreparable process task does the very similar job to repair process and attempt to repair the segments that are entered in the irreparableDB. Also Handle these secenarios:
1. If the irreparable segment, at the time of repair attempt is still irreparable, then exit and retry in the next cycle.
2. if the irreparable segment at the time of repair attempt just becomes reparable, then repair it immediately and delete the entry.

The assumption made is irreparableDB is generally small and has to be constantly checked hoping that any irreparable segment becomes repairable, just repair it as soon as possible. 

Implementing these Changes as above, makes least code changes without having to refactor. 

Bill, I am not clear about you suggestion. I can give you a call we can talk more in detail.

Reply:
@wthorp checker can't check a specific segment, but repairer can

Reply:
> 2. If the irreparable segment, at the time of repair attempt is still irreparable, then exit and retry in the next cycle.

@aligeti when this loop is broken?